src/etl/disease_etl.py

```diff
@@ -441,70 +464,89 @@ def get_generators(self, disease_data, batch_size, data_provider):
+
+                    if 'with' in disease_record:
+                        with_record = disease_record.get('with')
+                        for rec in with_record:
+                            disease_unique_key = disease_unique_key + rec
+                        for rec in with_record:
+                            with_map = {
+                                "diseaseUniqueKey": disease_unique_key,
                                 "withD": rec
                             }
-                            withs.append(withMap)
+                            withs.append(with_map)
 
-                    if disease_record.get('pgeIds') is not None:
-                        for pge in disease_record.get('pgeIds'):
-                            pge_map = {"pecjPrimaryKey": pecjPrimaryKey,
+                    if disease_allele.get('pgeIds') is not None:
+                        for pge in disease_allele.get('pgeIds'):
+                            pge_map = {"pecjPrimaryKey": pecj_primary_key,
                                        "pgeId": pge}
                             pge_list_to_yield.append(pge_map)
 
-                    if 'annotationDP' in disease_record:
-                        for adp in disease_record['annotationDP']:
-                            crossRefId = adp.get('crossRefId')
+                    if 'annotationDP' in disease_allele:
+                        for adp in disease_allele['annotationDP']:
+                            cross_ref_id = adp.get('crossRefId')
                             pages = adp.get('dpPages')
-                            annotationType = adp.get('annotationType')
+                            annotation_type = adp.get('annotationType')
                             local_crossref_id = ""
-                            prefix = crossRefId
-                            if annotationType is None:
-                                annotationType = 'curated'
+                            prefix = cross_ref_id
+                            if annotation_type is None:
+                                annotation_type = 'curated'
                             if pages is not None and len(pages) > 0:
                                 for page in pages:
-                                    modGlobalCrossRefId = ETLHelper.get_page_complete_url(local_crossref_id,
-                                                                                          self.xrefUrlMap, prefix, page)
-                                    xref = ETLHelper.get_xref_dict(local_crossref_id, prefix, page, page, crossRefId,
-                                                                   modGlobalCrossRefId, crossRefId + page + annotationType)
-                                    xref['dataId'] = diseaseUniqueKey
-                                    if 'loaded' in annotationType:
+                                    mod_global_cross_ref_id = ETLHelper.get_page_complete_url( \
```

Comment:
can the first item in the list be in line with the function call, I find this more readable than the \